packages/electron-updater/src/NsisUpdater.ts

```diff
@@ -31,7 +31,10 @@ export class NsisUpdater extends BaseUpdater {
       task: async (destinationFile, downloadOptions, packageFile, removeTempDirIfAny) => {
         const packageInfo = fileInfo.packageInfo
         const isWebInstaller = packageInfo != null && packageFile != null
-        if (isWebInstaller || (await this.differentialDownloadInstaller(fileInfo, downloadUpdateOptions, destinationFile, provider))) {
+        if (
+          (isWebInstaller && !downloadUpdateOptions.disableWebInstaller) ||
+          (await this.differentialDownloadInstaller(fileInfo, downloadUpdateOptions, destinationFile, provider))
+        ) {
```

Comment:
Thoughts on just checking if the web installer is supposed to be disabled first and throwing the error at that point? Then we don't need to throw an error any further down (lines 52-54)
```
if (isWebInstaller && downloadUpdateOptions.disableWebInstaller) {
  throw newError(
   `Unable to download new version ${downloadUpdateOptions.updateInfoAndProvider.info.version}. Web Installers are disabled`,
   "ERR_UPDATER_WEB_INSTALLER_DISABLED"
  )  
}
if (!isWebInstaller && !downloadUpdateOptions.disableWebInstaller) {
  this._logger.warn('disableWebInstaller is set to false, you should set it to true if you do not plan on using a web installer.')
}
if (isWebInstaller || (await this.differentialDownloadInstaller(fileInfo, downloadUpdateOptions, destinationFile, provider))) {
  ....
}
```

Reply:
Works for me!